atomix/cluster/src/test/java/io/atomix/raft/roles/PassiveRoleTest.java

```diff
@@ -81,43 +76,16 @@ public void setup() {
   }
 
   @Test
-  public void shouldRejectRequestIfDifferentNumberEntriesAndChecksums() {
+  public void shouldFailAppendWithIncorrectChecksum() {
     // given
-    final List<RaftLogEntry> entries = generateEntries(1);
-    final AppendRequest request = new AppendRequest(2, "", 1, 1, entries, List.of(1L, 2L), 1);
+    final List<PersistedRaftRecord> entries = new ArrayList<>();
+    entries.add(new PersistedRaftRecord(1, 1, 1, 1, new byte[1]));
+    final AppendRequest request = new AppendRequest(2, "", 1, 1, entries, 1);
 
     // when
     final AppendResponse response = role.handleAppend(request).join();
 
     // then
     assertThat(response.succeeded()).isFalse();
```

Comment:
The log is mocked and the check doesn't happen in the PassiveRole anymore so this test doesn't test that the append fails due to an incorrect checksum. It fails because the prevLogIndex is greater than the local value

Reply:
Oops :see_no_evil: Instead of mocking the log, we can provide an actual log. I don't see other way to test it.